src/main/java/com/minecolonies/api/colony/managers/interfaces/expeditions/ColonyExpedition.java

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param id               the id of the expedition.
+     * @param expeditionTypeId the expedition type.
+     * @param members          the members for this expedition.
+     * @param equipment        the list of equipment for this expedition.
+     */
+    public ColonyExpedition(
+      final int id,
+      final @NotNull ResourceLocation expeditionTypeId,
+      final @NotNull Map<Integer, IExpeditionMember<?>> members,
+      final @NotNull List<ItemStack> equipment)
+    {
+        super(members, equipment, List.of());
+        this.id = id;
+        this.expeditionTypeId = expeditionTypeId;
+    }
+
+    /**
+     * Deserialization constructor.
+     *
+     * @param members          the members for this expedition.
+     * @param equipment        the list of equipment for this expedition.
+     * @param results          the results for this expedition.
+     * @param id               the id of the expedition.
+     * @param expeditionTypeId the expedition type.
+     */
+    private ColonyExpedition(
+      final int id,
+      final @NotNull ResourceLocation expeditionTypeId,
+      final @NotNull List<IExpeditionMember<?>> members,
+      final @NotNull List<ItemStack> equipment,
+      final @NotNull List<ExpeditionStage> results)
+    {
+        super(members.stream().collect(Collectors.toMap(IExpeditionMember::getId, v -> v)), equipment, results);
+        this.id = id;
+        this.expeditionTypeId = expeditionTypeId;
+    }
+
+    /**
+     * Create a colony expedition instance from compound data.
+     *
+     * @param compound the compound data.
+     * @return the expedition instance.
+     */
+    @NotNull
+    public static ColonyExpedition loadFromNBT(final CompoundTag compound)
+    {
+        final int id = compound.getInt(TAG_ID);
+        final ResourceLocation expeditionTypeId = new ResourceLocation(compound.getString(TAG_EXPEDITION_TYPE));
+
+        return AbstractExpedition.loadFromNBT(compound, (members, equipment, results) -> new ColonyExpedition(id, expeditionTypeId, members, equipment, results));
+    }
+
+    /**
+     * Unique id for this expedition instance.
+     *
+     * @return the id.
+     */
+    public int getId()
+    {
+        return id;
+    }
+
+    /**
+     * Get the expedition type id for this expedition.
+     *
+     * @return the expedition type id.
+     */
+    @NotNull
+    public ResourceLocation getExpeditionTypeId()
+    {
+        return expeditionTypeId;
+    }
+
+    @Override
+    public void write(final CompoundTag compound)
+    {
+        super.write(compound);
+        compound.putInt(TAG_ID, id);
+        compound.putString(TAG_EXPEDITION_TYPE, expeditionTypeId.toString());
+    }
+
+    /**
+     * Get the expeditionary visitor that is leading this expedition.
+     *
+     * @return the member instance.
+     */
+    @NotNull
+    public ExpeditionVisitorMember getLeader()
+    {
+        for (final IExpeditionMember<?> member : members.values())
```

Comment:
Would store the leader in a seperate dara structure tbh. Leader + list of members